lib/srv/desktop/rdp/rdpclient/src/rdpdr/scard.rs

```diff
@@ -139,10 +142,17 @@ impl Client {
         Ok(Some(Box::new(resp)))
     }
 
-    fn handle_is_valid_context(&self, input: &mut Payload) -> RdpResult<Option<Box<dyn Encode>>> {
+    fn handle_is_valid_context(
+        &mut self,
+        input: &mut Payload,
+    ) -> RdpResult<Option<Box<dyn Encode>>> {
         let req = Context_Call::decode(input)?;
+        let code = match self.contexts.get_card_context(req.context.value) {
```

Comment:
This is the only functional change in this PR. Instead of reporting that all contexts are valid, we actually check to see if we have record of the context and respond accordingly.